java/vector/src/main/java/org/apache/arrow/vector/ipc/message/ArrowRecordBatch.java

```diff
@@ -106,6 +106,16 @@ private ArrowRecordBatch(boolean dummy, int length, List<ArrowFieldNode> nodes,
     this.buffersLayout = Collections.unmodifiableList(arrowBuffers);
   }
 
+  /**
+   * Returns the serialized form of {@link RecordBatch} wrapped in a {@link org.apache.arrow.flatbuf.Message}.
+   */
+  public ByteBuffer toSerializedFlatBuffer() {
```

Comment:
I'm not sure about adding this API. I don't think it is something the end user will call, and it's a little confusing that it's only the metadata. This is only used in Flight correct? Could this be added to `MessageSerializer` instead or change `MessageSerializer.serialize()` to create a `FlatBufferBuilder` itself?

Reply:
moving to serializer

Reply:
I also moved all the other toSerializedFlatbuffer additions to MessageSerializer (will push new commit shortly).